x-pack/plugins/cases/server/services/so_references.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObject, SavedObjectReference } from '@kbn/core/types';
+import { isEqual, uniqWith } from 'lodash';
+import {
+  CommentAttributesWithoutSORefs,
+  CommentRequest,
+  CommentAttributes,
+  CommentPatchAttributes,
+} from '../../common/api';
+import { PersistableStateAttachmentTypeRegistry } from '../attachment_framework/persistable_state_registry';
+import {
+  injectPersistableReferencesToSO,
+  extractPersistableStateReferencesFromSO,
+} from '../attachment_framework/so_references';
+import { EXTERNAL_REFERENCE_REF_NAME } from '../common/constants';
+import { isCommentRequestTypeExternalReferenceSO } from '../common/utils';
+import { SOReferenceExtractor } from './so_reference_extractor';
+
+export const getAttachmentSOExtractor = (attachment: Partial<CommentRequest>) => {
+  const fieldsToExtract = [];
+
+  if (isCommentRequestTypeExternalReferenceSO(attachment)) {
+    fieldsToExtract.push({
+      path: 'externalReferenceId',
+      type: attachment.externalReferenceStorage.soType,
+      name: EXTERNAL_REFERENCE_REF_NAME,
+    });
+  }
+
+  return new SOReferenceExtractor(fieldsToExtract);
+};
+
+export const injectAttachmentSOReferences = (
+  soExtractor: SOReferenceExtractor,
+  savedObject: SavedObject<CommentAttributesWithoutSORefs>,
+  persistableStateAttachmentTypeRegistry: PersistableStateAttachmentTypeRegistry
+) => {
```

Comment:
Just to clarify my statements about `getAttachmentSOExtractor`, I'm suggesting calling it here (but leave it as a separate function because we also have to call it to create an soExtractor to repopulate the patch's output).

Reply:
You made me understand that I did not `inject` the refs to the attributes on patches. I fixed it and move the `getAttachmentSOExtractor` inside the inject/extract methods.